app-header/app-header.html

```diff
@@ -396,42 +427,67 @@
         if (this._height === 0) {
           return;
         }
-        var progress;
-        var mayHeaderMove = !this.fixed || this.condenses;
-        var top = Math.min(this._headerDy, Math.max(0,
-            this.reveals ? this._top + scrollTop - this._lastScrollTop : scrollTop
-        ));
-
-        if (this._shouldCondense(scrollTop)) {
-          top = Math.max(this._deltaHeight, top);
+
+        var progress = 0;
+        var top = 0;
+        var lastTop = this._top;
+        var lastScrollTop = this._lastScrollTop;
+        var headerDy = this._headerDy;
+        var deltaScrollTop = scrollTop - this._lastScrollTop;
+        var isScrollingDown = scrollTop > this._lastScrollTop;
+
+        if (this._mayMove()) {
+          top = Math.min(headerDy, Math.max(0,
+              this.reveals ? lastTop + deltaScrollTop : scrollTop
+          ));
+        }
+
+        if (this.snaps && this.reveals && !this.disabled && Math.abs(deltaScrollTop) < 100) {
+          if (isScrollingDown && scrollTop >= this._headerDy) {
+
+            top = headerDy;
+            this.style.transition = this._snapTransition;
+
+          } else if (!isScrollingDown && scrollTop >= this._deltaHeight) {
+
+            top = this.condenses ? this._deltaHeight : 0;
+            this.style.transition = this._snapTransition;
+
+          } else {
+            this.style.transition = '';
+          }
+        } else if (scrollTop >= this._deltaHeight) {
+
+          top = this.condenses ? Math.max(this._deltaHeight, top) : top;
+          this.style.transition = '';
+
         }
+
         if (this._deltaHeight === 0) {
-          progress = scrollTop > 0 ? 1 : 0;
+          progress = scrollTop > 0 ? 1 : 0
         } else {
           progress = top / this._deltaHeight;
         }
-        if (forceUpdate || progress !== this._progress || top !== this._top || scrollTop === 0) {
-          this._runEffects(progress, mayHeaderMove ? top : 0);
-          this._transformHeader(mayHeaderMove ? top : 0);
-          this.fire('app-header-transform', { top: top, progress: progress }, { bubbles: false });
```

Comment:
We have a few demos (e.g. app-header/demo/contacts.html, app-header-layout/demo/music.html) depend on `app-header-transform`.  Can you update those demos to use the new way?